docs/cookbook/extending-product-list.md

```diff
@@ -84,23 +198,24 @@ use Shopsys\ReadModelBundle\Product\Listed\ListedProductView as BaseListedProduc
 class ListedProductViewFactory extends BaseListedProductViewFactory
 {
     /**
-     * @param \Shopsys\ShopBundle\Model\Product\Product $product
+     * @param array $hit
      * @param \Shopsys\ReadModelBundle\Image\ImageView|null $imageView
      * @param \Shopsys\ReadModelBundle\Product\Action\ProductActionView $productActionView
+     * @param \Shopsys\FrameworkBundle\Model\Pricing\Group\PricingGroup $pricingGroup
      * @return \Shopsys\ReadModelBundle\Product\Listed\ListedProductView
      */
-    public function createFromProduct(Product $product, ?ImageView $imageView, ProductActionView $productActionView): BaseListedProductView
+    public function createFromHit(array $hit, ?ImageView $imageView, ProductActionView $productActionView, PricingGroup $pricingGroup): ListedProductView
     {
         return new ListedProductView(
-            $product->getId(),
-            $product->getName(),
-            $product->getShortDescription($this->domain->getId()),
-            $product->getCalculatedAvailability()->getName(),
-            $this->productCachedAttributesFacade->getProductSellingPrice($product),
-            $this->getFlagIdsForProduct($product),
+            (int)$hit['_id'],
+            $hit['_source']['name'],
+            $hit['_source']['shortDescription'],
```

Comment:
Right here I said to myself that I don't want to tell users to do this 😄
Please consider deeply the comments about hits and valid structure in read-model package